src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function add(a) {
+    if (a !== undefined) {
+      sum += a;
+
+      return add;
+    } else {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
```

Comment:
When you are using return in if statement you could remove else statement
```suggestion
    if (a !== undefined) {
      sum += a;

      return add;
    } 
      const result = sum;

      sum = 0;

      return result;
    
```